mindmeld/converter/dialogflow.py

```diff
@@ -0,0 +1,375 @@
+                      '@sys.duration': 'sys_duration',
+                      '@sys.number': 'sys_number',
+                      '@sys.cardinal': 'sys_number',
+                      '@sys.ordinal': 'sys_ordinal',
+                      '@sys.unit-currency': 'sys_amount-of-money',
+                      '@sys.unit-volume': 'sys_volume',
+                      '@sys.email': 'sys_email',
+                      '@sys.phone-number': 'sys_phone-number',
+                      '@sys.url': 'sys_url'}
+
+    def __init__(self, dialogflow_project_directory, mindmeld_project_directory):
+        self.dialogflow_project_directory = dialogflow_project_directory
+        self.mindmeld_project_directory = mindmeld_project_directory
+        self.directory = os.path.dirname(os.path.realpath(__file__))
+        self.entities_list = set()
+        self.intents_list = set()
+
+    def create_mindmeld_directory(self):
+        self.create_directory(self.mindmeld_project_directory)
+        self.create_directory(os.path.join(self.mindmeld_project_directory, "data"))
+        self.create_directory(os.path.join(self.mindmeld_project_directory, "domains"))
+        self.create_directory(os.path.join(self.mindmeld_project_directory, "domains",
+                                           "general"))
+        self.create_directory(os.path.join(self.mindmeld_project_directory, "entities"))
+
+    # =========================
+    # create training data (entities, intents)
+    # =========================
+
+    def _create_entities_directories(self, entities):
+        """ Creates directories + files for all languages/files.
+        Currently does not use meta data in entityName.json files (the keys in var entities).
+        """
+        for languages in entities.values():
+            for sub in languages.values():
+                dialogflow_entity_file = os.path.join(self.dialogflow_project_directory,
+                                                      "entities", sub + ".json")
+
+                mindmeld_entity_directory_name = self.clean_check(sub, self.entities_list)
+
+                mindmeld_entity_directory = os.path.join(self.mindmeld_project_directory,
+                                                         "entities",
+                                                         mindmeld_entity_directory_name)
+
+                self.create_directory(mindmeld_entity_directory)
+
+                self._create_entity_file(dialogflow_entity_file, mindmeld_entity_directory)
+
+    @staticmethod
+    def _create_entity_file(dialogflow_entity_file, mindmeld_entity_directory):
+        source_en = open(dialogflow_entity_file, 'r')
+        target_gazetteer = open(os.path.join(mindmeld_entity_directory, "gazetteer.txt"), 'w')
+        target_mapping = open(os.path.join(mindmeld_entity_directory, "mapping.json"), 'w')
+
+        datastore = json.load(source_en)
+        mapping_dict = {"entities": []}
+
+        for item in datastore:
```

Comment:
Nit: We could templatize this code block to append dictionary objects to the file

Reply:
I'm adding dictionary objects to a list, where the list is the value to the key "entities" in mapping_dict, and dumping mapping_dict at the end. At the moment I can't think of clean solution to append these objects to the file one at a time instead since you would be adding to the end of the file and not to the above mentioned list.